### PR TITLE
Migrate DropDownMenu; closes #2351.

### LINKED ISSUE #2351: Migrate DropDownMenu
- migrate/copy logic from `assets\ZZZ\components\ui\drop-down-menu` to `assets\src\application\ui\layout`

- replace WP component with AntD equivalent, maybe: https://ant.design/components/dropdown/

- ensure that keyboard navigation works (if not then revert to using WP component)

### PR DESCRIPTION
**Pull request by nicolad**
## Problem this Pull Request solves

-  Migrates over the GB's DropdownMenu (a copy of LICENCE is also added): https://github.com/WordPress/gutenberg/tree/master/packages/components/src/dropdown-menu
- Closes: https://github.com/eventespresso/event-espresso-core/issues/2351

## How has this been tested
Peek 2020-03-12 15-35

### CODE REVIEW

File: js-green-licenses.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
 		"gaxios",
 		"js-green-licenses",
 		"jsonify",
+		"mousetrap",
```
Comment (nicolad) [inline js-green-licenses.json]: LICENCE: https://github.com/ccampbell/mousetrap/blob/master/LICENSE

File: assets/src/application/ui/layout/EspressoDropdownMenu/index.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import classnames from 'classnames';
+import { flatMap, isEmpty, isFunction } from 'lodash'; // to be replaced with ramda
+import { DOWN } from '@wordpress/keycodes';
+import { Dropdown, NavigableMenu } from '@wordpress/components';
```
Comment (nicolad) [inline assets/src/application/ui/layout/EspressoDropdownMenu/index.tsx]: This still depends on some other packages.

File: assets/src/application/ui/layout/EspressoDropdownMenu/style.scss [hunk 3]
```diff
@@ -0,0 +1,489 @@
+.components-dropdown {
```
Comment (nicolad) [inline assets/src/application/ui/layout/EspressoDropdownMenu/style.scss]: styles from this file are being used from `@wordpress/*`

File: assets/src/application/ui/layout/EspressoDropdownMenu/style.scss [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+		padding-left: 2rem;
+
+		&.components-button {
+			height: auto;
+			padding-left: 2rem;
+		}
+	}
+}
+
+.components-popover {
```
Comment (nicolad) [inline assets/src/application/ui/layout/EspressoDropdownMenu/style.scss]: For some reasons, styles from popover were not automatically imported, so I had to add these manually.

File: assets/src/application/ui/layout/EspressoDropdownMenu/types.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+export interface DropdownMenuProps {
+	children(props: AnyObject): JSX.Element;
+	className?: string;
+	controls?: [];
+	icon?: string | null;
+	menuProps?: AnyObject;
+	label: string;
+	popoverProps?: AnyObject;
+	toggleProps?: AnyObject;
+	[key: string]: any;
```
Comment (manzoorwanijk) [inline assets/src/application/ui/layout/EspressoDropdownMenu/types.ts]: You could use `DropdownMenuProps  extends AnyObject` and remove this line here

Comment (nicolad) [reply assets/src/application/ui/layout/EspressoDropdownMenu/types.ts]: Done, thx TS-master :bowing_man:

**Comment by nicolad** [comment]
I will continue improving this upon integration into date and ticket cards.

File: assets/src/application/ui/layout/EspressoDropdownMenu/LICENSE.md [hunk 6]
```diff
@@ -0,0 +1,394 @@
+### WordPress - Web publishing software
+
+    Copyright 2011-2020 by the contributors
```
Comment (tn3rb) [inline assets/src/application/ui/layout/EspressoDropdownMenu/LICENSE.md]: PERFECT !!!  💯

File: assets/src/application/ui/layout/EspressoDropdownMenu/index.tsx (hunk 2 above)
Comment (tn3rb) [reply assets/src/application/ui/layout/EspressoDropdownMenu/index.tsx]: ya that kinda stinks, but whatcha gonna do?
hopefully we can utilize a better solution sometime soon

File: assets/src/application/ui/layout/EspressoDropdownMenu/style.scss [hunk 7]
```diff
@@ -0,0 +1,489 @@
...
+	font-size: 13px;
+	line-height: 1.4;
+
+	.components-dropdown-menu__menu-item,
+	.components-menu-item {
+		background-color: transparent !important;
+		border: none !important;
+		box-shadow: unset !important;
+		cursor: pointer;
+		margin: 0 !important; // important was needed because it is conflicting with EspressoButton styles
```
Comment (tn3rb) [inline assets/src/application/ui/layout/EspressoDropdownMenu/style.scss]: `!important` === necessary evil

File: assets/src/application/ui/layout/EspressoDropdownMenu/style.scss [hunk 8]
```diff
@@ -0,0 +1,489 @@
...
+		&.components-button {
+			height: auto;
+			padding-left: 2rem;
+		}
+	}
+}
+
+.components-popover {
+	position: fixed;
+	z-index: 1000000;
```
Comment (tn3rb) [inline assets/src/application/ui/layout/EspressoDropdownMenu/style.scss]: ug... more necessary evil although this should have been more preventable (not referring to you at all, just the greater dev community)

**Pull request merged**